packages/astro/src/astro-components/unstyled/SignInButton.astro

```diff
@@ -0,0 +1,49 @@
+---
+import type { HTMLTag, Polymorphic } from 'astro/types'
+import type { SignInProps } from "@clerk/types";
+type Props<Tag extends HTMLTag> = Polymorphic<SignInProps & { as: Tag; id?: string; mode?: 'redirect' | 'modal' }>
+
+import { customAlphabet, urlAlphabet } from "nanoid";
+
+const safeId = customAlphabet(urlAlphabet, 10)();
+
+const {
+  as: Tag = 'button',
+  id = safeId,
+  forceRedirectUrl,
+  fallbackRedirectUrl,
+  signUpFallbackRedirectUrl,
+  signUpForceRedirectUrl,
+  mode,
+  ...elementProps
+} = Astro.props
+
+const signInOptions = {
+  forceRedirectUrl,
+  fallbackRedirectUrl,
+  signUpFallbackRedirectUrl,
+  signUpForceRedirectUrl,
+};
+---
+
+<Tag {...elementProps} id={id}>
+  <slot>Sign in</slot>
+</Tag >
```

Comment:
For Astro components, we're not using the `asChild` and children props to render a custom element like what we have in the React SDK. We're using the built-in slots. More info on ECO-2 ticket